staging/src/k8s.io/apiserver/pkg/authentication/request/x509/x509.go

```diff
@@ -73,6 +73,10 @@ func (a *Authenticator) AuthenticateRequest(req *http.Request) (user.Info, bool,
 
 	chains, err := req.TLS.PeerCertificates[0].Verify(optsCopy)
 	if err != nil {
+		switch err.(type) {
+		case x509.UnknownAuthorityError:
```

Comment:
It is at least a fair point that the UnionDon'tFailOnError thingy should have done the right thing here already, so why isn't it? @cheftako